base/applications/cmdutils/tasklist/tasklist.c

```diff
@@ -0,0 +1,338 @@
+/*
+ * PROJECT:     ReactOS Tasklist Command
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Displays a list of currently running processes on the computer.
+ * COPYRIGHT:   Copyright 2020 He Yang (1160386205@qq.com)
+ */
+
+#include "tasklist.h"
+
+// the strings in opList are the command-line options.
+// should always correspond with the defines below, in sequence (except OP_PARAM_INVALID)
+static PCWSTR opList[] = { L"?", L"nh" };
+
+#define OP_PARAM_INVALID    -1
+#define OP_PARAM_HELP       0
+#define OP_PARAM_NOHEADER    1
+
+// the max string length PrintResString can handle
+#define RES_STR_MAXLEN 64
+
+// Print split line
+VOID PrintSplitLine(UINT Length)
+{
+    for (; Length; Length--)
+    {
+        ConPuts(StdOut, L"=");
+    }
+}
+
+// Print spaces
+VOID PrintSpace(UINT Length)
+{
+    ConPrintf(StdOut, L"%*ls", (INT)Length, L"");
+}
+
+// Print a string.
+// if bAlignLeft == TRUE then aligned to left, otherwise aligned to right
+// MaxWidth is the width for printing.
+INT PrintString(LPCWSTR String, INT MaxWidth, BOOL bAlignLeft)
+{
+    return ConPrintf(StdOut, bAlignLeft ? L"%-*.*ls" : L"%*.*ls", MaxWidth, MaxWidth, String);
```

Comment:
Maybe casts are needed now for MaxWidth? (Does the compiler give warnings if they stay UINT?)

Reply:
compiler gives no warnings, but the document (printf's) says if it's negative, it will be ignored. 🤣
(IMO, even if it's not negative, like 0x7fffffff, it might already is a disaster)

Reply:
possibly just left it here and add a comment saying this function will ignore this param if it's negative?

Reply:
Then you can just go ahead and convert all your width variables to unsigned ones. This will make the code self-documented (namely that no width can be negative)

Reply:
ok!